pkg/operatorsource/configuring_phase_reconciler.go

```diff
@@ -0,0 +1,120 @@
+package operatorsource
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	catalogsourceconfigPrefix = "opsrc"
+)
+
+// NewConfiguringPhaseReconciler returns a PhaseReconciler which reconciles
+// an OperatorSource object in "Configuring" phase.
+func NewConfiguringPhaseReconciler(datastore Datastore, kubeclient Kubeclient) PhaseReconciler {
+	return &configuringPhaseReconciler{
+		datastore:  datastore,
+		kubeclient: kubeclient,
+	}
+}
+
+// Given a name of OperatorSource object, this function returns the name
+// of the corresponding CatalogSourceConfig type object
+func getCatalogSourceConfigName(operatorsourceName string) string {
+	return fmt.Sprintf("%s-%s", catalogsourceconfigPrefix, operatorsourceName)
+}
+
+// configuringPhaseReconciler is used to reconcile an OperatorSource object which is in "Configuring" phase.
+//
+// Reconcile reconciles an object of OperatorSource type which is currently in "Configuring" phase.
+// It ensures that a corresponding CatalogSourceConfig object exists.
+// Upon success, it returns "Succeeded" as the next desired phase.
+// On error, the function returns "Failed" as the next desied phase, and Message reflects the error message.
+// If the corresponding CatalogSourceConfig object already exists, no further action is taken.
+type configuringPhaseReconciler struct {
+	datastore  Datastore
+	kubeclient Kubeclient
+}
+
+func (r *configuringPhaseReconciler) Reconcile(ctx context.Context, in *v1alpha1.OperatorSource) (out *v1alpha1.OperatorSource, status *NextPhase, err error) {
+	logger := getLoggerWithOperatorSourceTypeFields(in)
```

Comment:
good point; and yes there is - by use of `context`. we can use `context` to pass along out of band request scoped data (logger with fields in this case). 

In the `Handle` function, we create the `logger` and inject it
```
ctx := context.WithValue(ctx, "logger," logger)
```

From `Reconcile` where we need to use it, we retrieve it
```
logger, ok := ctx.Value("logger").(*logrus.Logger)
```

Reply:
This makes sense. @ravisantoshgudimetla what is your take on this approach?

Reply:
As I told earlier, I am kind of against using context. In this case, I think, it is better to have a custom logging function(or a var) in a new package called logging and import that fn instead of passing logger in a context.

Reply:
The logger here is `event scoped`, we need to initialize it when we receive an event from the sdk. A global variable, I am afraid, won't solve our problem, unless we abandon this pattern and stick to a default logger ( without structured tags ) for everything.

Reply:
Looks like both (context and global variables) are discouraged.

https://dave.cheney.net/2017/01/23/the-package-level-logger-anti-pattern
https://dave.cheney.net/2017/01/26/context-is-for-cancelation